middle_end/compilenv.ml

```diff
@@ -253,59 +211,58 @@ let toplevel_approx :
   (string, Clambda.value_approximation) Hashtbl.t = Hashtbl.create 16
 
 let record_global_approx_toplevel () =
-  Hashtbl.add toplevel_approx current_unit.ui_name
+  Hashtbl.add toplevel_approx
+    (CU.Name.to_string (CU.name current_unit.ui_name))
     (get_clambda_approx current_unit)
 
 let global_approx id =
   if Ident.is_predef id then Clambda.Value_unknown
   else try Hashtbl.find toplevel_approx (Ident.name id)
   with Not_found ->
+    (* XXX Does this need anything to do with packs on it? *)
```

Comment:
_Does_ it need anything to do with packs? Are packs reflected in Clambda code?

Reply:
I think this XXX comment can be deleted actually.  Presumably in stage 2 this is going to use `CU.Name.t` anyway, because `Lambda` won't be using `Ident.t` for globals any more?

Reply:
`CU.t`, actually, so the answer will be emphatically "no, we don't need to do anything with packs." Deleted the comment.